core/chaincode/shim/chaincode.go

```diff
@@ -572,11 +572,11 @@ func (stub *ChaincodeStub) GetRows(tableName string, key []Column) (<-chan Row,
 	if err != nil {
 		return nil, fmt.Errorf("Error fetching rows: %s", err)
 	}
-	defer iter.Close()
 
 	rows := make(chan Row)
 
 	go func() {
+		defer iter.Close()
 		for iter.HasNext() {
 			_, rowBytes, err := iter.Next()
 			if err != nil {
```

Comment:
@ghaskins Good point. In the chaincode log, an error message which says "Another state request pending for this Uuid. Cannot process." is recorded. It looks the code is checking and make sure that there is only one channel per uuid. I don't know why.